v-next/hardhat-ignition/src/internal/tasks/deploy.ts

```diff
@@ -14,16 +35,289 @@ interface TaskDeployArguments {
   writeLocalhostDeployment: boolean;
 }
 
+const log = debug("hardhat:ignition");
+
 const taskDeploy: NewTaskActionFunction<TaskDeployArguments> = async (
-  deployArgs,
-  _hre: HardhatRuntimeEnvironment,
+  {
+    modulePath,
+    parameters: parametersInput,
+    deploymentId: givenDeploymentId,
+    defaultSender,
+    reset,
+    verify,
+    strategy: strategyName,
+    writeLocalhostDeployment,
+  },
+  hre: HardhatRuntimeEnvironment,
 ): Promise<void> => {
-  if (deployArgs.verify) {
+  if (verify) {
     throw new HardhatError(HardhatError.ERRORS.INTERNAL.NOT_IMPLEMENTED_ERROR, {
       message:
         "Verifying deployments is not available yet. It will be available in a future version of the Harhdat 3 Alpha",
     });
+
+    // TODO: Bring back with the port of hardhat-verify
+    // if (
+    //   hre.config.etherscan === undefined ||
+    //   hre.config.etherscan.apiKey === undefined ||
+    //   hre.config.etherscan.apiKey === ""
+    // ) {
+    //   throw new HardhatError(
+    //     HardhatError.ERRORS.IGNITION.ETHERSCAN_API_KEY_NOT_CONFIGURED,
+    //   );
+    // }
+  }
+
+  const connection = await hre.network.connect();
+
+  const chainId = Number(
+    await connection.provider.request({
+      method: "eth_chainId",
+    }),
+  );
+
+  const deploymentId = resolveDeploymentId(givenDeploymentId, chainId);
+
+  const deploymentDir =
+    connection.networkName === "hardhat" && !writeLocalhostDeployment
+      ? undefined
+      : path.join(hre.config.paths.ignition, "deployments", deploymentId);
```

Comment:
To be revisited after the alpha

Reply:
Do you mean how we setup the deployment directory, or the place we are calculating and setting the deployment dir - as in, this should be done during config resolution in a hook?

Reply:
We should change the network name check, to a check on network type - EDR networks are ephemeral.